### PR TITLE
Support YAML config options

### PR DESCRIPTION
**Pull request by mtojek**
Changes:
- support YAML fields
- determine if field is required
- refresh beats

### CODE REVIEW

**Comment by mtojek** [comment]
Redis key `key.pattern` option added in https://github.com/elastic/beats/pull/17875

File: dev/packages/beats/apache-0.0.1/manifest.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ categories:
 - logs
 - metrics
 release: ga
+removable: false
```
Comment (ruflin) [inline dev/packages/beats/apache-0.0.1/manifest.yml:12]: This should be true as the default is true. I suspect an error in the package code?

Comment (mtojek) [reply dev/packages/beats/apache-0.0.1/manifest.yml]: Fixed.

File: dev/packages/beats/system-0.0.1/manifest.yml [hunk 2]
```diff
@@ -9,6 +9,7 @@ categories:
 - logs
 - metrics
 release: ga
+removable: true
```
Comment (ruflin) [inline dev/packages/beats/system-0.0.1/manifest.yml:12]: Nit: This is where it is false. But I think we can do this as a manual adjustment.

Comment (mtojek) [reply dev/packages/beats/system-0.0.1/manifest.yml]: Fixed.

If this is only for the system package, I guess that this option could be inverted - single property for non-removable/locked/etc.

**Pull request merged**